File: Wiki/C/code/C_Programming_A_Modern_Approach/ch06_Loops/exercises/02.c
// Purpose: compute output of a given program fragment, which uses a `do` loop
// Reference: page 121 (paper) / 146 (ebook)

#include <stdio.h>

    int
main(void)
{
    int i = 9384;
    do
    {
        printf("%d ", i);
        i /= 10;
    } while (i > 0);
    //     9384 938 93 9

    return 0;
}

File: Wiki/lua/code/Programming_in_Lua_3rd_edition/ch01_Getting_Started/exercises/02.lua
-- Purpose: Run the  `twice` example,  both by  loading the  file with  the `-l`
-- option and with `dofile()`.  Which way do you prefer?
-- Reference: page 8 (paper) / 27 (ebook)

local function twice(x)
  return 2 * x
end

-- With `-l`:{{{
--
-- Write the code in `/tmp/test/lua.lua`, then:
--
--     $ cd /tmp
--     $ mkdir test
--     $ lua -l test.lua
--     > print(twice(5))
--     10
--}}}
-- With `dofile()`:{{{
--
-- Write the code in `/tmp/test.lua`, then:
--
--     $ cd /tmp
--     $ lua
--     > dofile('/tmp/test.lua')
--     > print(twice(5))
--     10
--}}}

-- I  prefer `dofile()`  because  it doesn't  require  to put  the  script in  a
-- directory of `package.path`.
